feat(remote/aws): support AWS Secrets Manager as remote component (#718)
grafana/alloy: docs/sources/reference/components/remote.aws.secrets_manager.md

---
canonical: https://grafana.com/docs/alloy/latest/reference/components/remote.aws.secrets_manager/
description: Learn about remote.aws.secrets_manager
title: remote.aws.secrets_manager
---

# remote.aws.secret_manager

> `remote.aws.secrets_manager` securely exposes value of secrets located in [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) to other components.

Reviewer: ```suggestion
`remote.aws.secrets_manager` securely exposes the secrets located in [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) to other components.
```

> By default, the secret would be fetched one time only at startup. If configured, the secret will be polled for changes so that the most recent value is always available. 

Reviewer: ```suggestion
By default, the secret is fetched once only at startup. If configured, the secret is polled for changes so that the most recent value is always available. 
```

> Beware that this could incur cost due to frequent API calls.

Reviewer: ```suggestion
{{< admonition type="note" >}}
The polling for changes could incur costs due to frequent API calls.
{{< /admonition >}}
```

Are the costs monetary, or throttling, or something else?  Are the costs specific/identifiable?

Author: Monetary cost

> Multiple `remote.aws.secrets_manager` components can be specified using different name
> labels. By default, [AWS environment variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html) are used to authenticate against AWS. The `key` and `secret` arguments inside `client` blocks can be used to provide custom authentication.

Reviewer: ```suggestion
You can specify multiple `remote.aws.secrets_manager` components by using a unique label for each component.
By default, [AWS environment variables][] are used to authenticate against AWS.
For custom authentication, you can use the `key` and `secret` arguments inside `client` blocks.

 [AWS environment variables]: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
```

## Usage

```alloy
remote.aws.secrets_manager "LABEL" {
  id = AWS_SECRETS_MANAGER_SECRET_ID
}
```

## Arguments

The following arguments are supported:

Name             | Type       | Description                                                              | Default | Required
-----------------|------------|--------------------------------------------------------------------------|---------|---------
`id`             | `string`   | Secret ID.                                                               |         | yes
`poll_frequency` | `duration` | How often to poll the API for changes.                                   |         | no

## Blocks

Hierarchy | Name       | Description                                        | Required
----------|------------|----------------------------------------------------|---------

> client    | [client][] | Additional options for configuring the AWS client. | no

Reviewer: ```suggestion
client    | [client][] | Additional AWS client configuration options. | no
```

`key`            | `string` | Used to override default access key.                                                    |         | no
`secret`         | `secret` | Used to override default secret value.                                                  |         | no
`disable_ssl`    | `bool`   | Used to disable SSL, generally used for testing.                                        |         | no
`region`         | `string` | Used to override default region.                                                        |         | no
`signing_region` | `string` | Used to override the signing region when using a custom endpoint.                       |         | no

## Exported fields

The following fields are exported and can be referenced by other components:

Name      | Type                 | Description                
----------|----------------------|---------------------------------------------------------
`data`    | `map(secret)`        |  Data from the secret obtained from AWS Secrets Manager.

The `data` field contains a mapping from data field names to values.

## Component health

> Instances of `remote.aws.secrets_manager` report as healthy if most recent fetch of stored secrets was successful.

Reviewer: ```suggestion
Instances of `remote.aws.secrets_manager` report as healthy if the most recent fetch of stored secrets was successful.
```
